src/tasks/module.rs

```diff
@@ -0,0 +1,146 @@
+//! A module is an executable, that accepts structured
+//! data as Input (json) and performs
+//! actions based on the provided Data
+//!
+//! Task takes default data from <data-file>
+//! default-values get overwritten with custom values in
+//! host var host.vars.module_<module_name>
+//! default.toml should contain all information of the
+//! expected data structure
+
+use crate::prelude::*;
+
+use serde_json::{json, Value};
+use ssh2::Channel;
+
+use std::io::prelude::*;
+use std::path::Path;
+use std::fs::{File, self};
+
+/// Describe an `module` task
+pub struct Task {
+    data_path: Option<String>, // todo! implement with &str and livetime
+    module_path: String,
+    module_name: String,
+}
+
+
+impl Task {
+    /// Create a new `module` task
+
+    pub fn new(data_path: Option<String>, module_path: String) -> Self {
+        let module_name = module_path.split("/").last().unwrap().to_owned(); // todo better error handling
+        Self { data_path, module_path, module_name }
+    }
+}
+
+impl GenericTask<String> for Task {
+    fn prepare(&self, host: Host) -> Result<String> {
+
+        let hostvars = host.vars.clone();
+        
+        let default = json!({});
+
+        
+        let host_var_data: &Value = hostvars.get(&format!("module_{}", self.module_name)).unwrap_or(&default);
+        println!("host_var_data: {:?}", host_var_data);
```

Comment:
Don't forget to remove the debugging `println!` once you're done :)

Reply:
i hope they are all gone now